NAS2D/Filesystem.cpp

```diff
@@ -77,25 +70,15 @@ enum MountPosition
 };
 
 
-Filesystem::Filesystem(const std::string& argv_0, const std::string& appName, const std::string& organizationName) :
+Filesystem::Filesystem(const std::string& /*argv_0*/, const std::string& appName, const std::string& organizationName) :
```

Comment:
You now have a useless argument *at the front* of the argument list. Recommend flat out removing it.

`Filesystem(const std::string& appName, const std::string& organizationName)`

or, if you are afraid it will break something (why? You are specifically refactoring out old/broken code), use a delegating constructor:

```
Filesystem::Filesystem(const std::string& appName, const std::string& organizationName) : 
mBasePathSdlString{sdlString{SDL_GetBasePath()}.get()},
mPrefPath{SdlString{SDL_GetPrefPath(organizationName.c_str(), appName.c_str())}.get()}
{
}

Filesystem::Filesystem(const std::& /*argv_0*/, const std::string& appName, const std::string& organizationName) : 
Filesystem(appName, organizationName)
{
}
```

Reply:
Well that's exactly why I left it in. It would break source code compatibility if I removed that argument. I'm sure it will be removed eventually, just as part of a future change set. I wanted to do the initial refactor without changing the external API.

The delegating constructor idea may be useful for a transition period.